### ISSUE TITLE
[bugs] on ACN_CustomControlNetWeightsFlux": "ControlNet Custom Weights [Flux] 🛂🅐🅒🅝

### ISSUE THREAD

**Issue by George0726** [Kosinkadink/ComfyUI-Advanced-ControlNet#175]
Hi author,
Thanks for your contribution! However, I found there was something wrong with the FLUX custom weight node.
it seems like only the last two layers **(layer17, layer18)** make a contribution to results. For example, here I set all weights to 0 except for the last two layers. However, the values added to the FLUX model are not 0.

![a1f53d87-af44-4e12-8e40-ebb115f19581](https://github.com/user-attachments/assets/aeadd77a-52af-4acc-b1d2-6e1c0f601098)

![52f6389c-6408-49cb-a458-70b542d9fcbc](https://github.com/user-attachments/assets/63eeb5da-981e-4492-9ee2-ca69cad3223a)

**Comment by George0726** [comment]
FLUX controlnets duplicate control values 9 times. hence this doesn't work. **applied_to = set()**

-            applied_to = set()
+            # applied_to = set()
             for i in range(len(control_output)):
                 x = control_output[i]
                 if x is not None:
                     if self.global_average_pooling:
                         x = torch.mean(x, dim=(2, 3), keepdim=True).repeat(1, 1, x.shape[2], x.shape[3])

**Comment by Kosinkadink** [comment]
Thanks for the heads up! I'll look into patching that today

**Comment by George0726** [comment]
> Thanks for the heads up! I'll look into patching that today

Thank!
I discovered that the original ComfyUI code repeats the tensors ten times in line #116 of the comfy/ldm/flux/controlnet.py file. However, these repeated tensors share the same memory, which prevents independent modification of their values. If we want to customize the control weights for all 19 layers, we should consider adding the clone() operator to ensure that each tensor is a distinct copy, allowing for independent modifications in line #877 of custom_nodes/ComfyUI-Advanced-ControlNet/adv_control/utils.py.

**Comment by Kosinkadink** [comment]
A fix for this is now on the rework-modelpatcher branch, and will be merged in as soon as my ComfyUI hooks PR gets merged in (current ETA is around US Thanksgiving.
